### PR TITLE
[humanparser] Fixed typing of first name return

### PR DESCRIPTION
**Pull request by viralpickaxe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [unit test in package](https://github.com/ralyodio/humanparser/blob/master/test/index.js#L175-L181)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by MichalPodeszwa** [comment]
@viralpickaxe Love the $ExpectType.
I'm not sure the change in types is correct though.
Looking at nameparser tests seems like `firstName`, `lastName` and `fullName` are always returned, just sometimes is `""` without any value.

File: types/humanparser/index.d.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 declare namespace humanparser {
     interface NameOutput {
-        firstName: string;
+        firstName?: string | undefined;
```
Comment (MichalPodeszwa) [inline types/humanparser/index.d.ts:8]: According to tests here: https://github.com/ralyodio/humanparser/blob/master/test/index.js#L214 This should be `string` if I'm not mistaken

**Review by MichalPodeszwa — CHANGES_REQUESTED** [review]

File: types/humanparser/index.d.ts (hunk 1 above)
Comment (viralpickaxe) [reply types/humanparser/index.d.ts:8]: Yep I think that's what we all expect, but it appears that in cases where just a title + last name are passed in `firstName` is not present in the result and this looks intentional according to this [unit test](https://github.com/ralyodio/humanparser/blob/master/test/index.js#L175-L181)

Comment (MichalPodeszwa) [reply types/humanparser/index.d.ts:8]: Thank you - did not expect to find that, after seeing the test for empty string

**Review by MichalPodeszwa — APPROVED** [review]
LGTM!

File: types/humanparser/index.d.ts (hunk 1 above)
Comment (viralpickaxe) [reply types/humanparser/index.d.ts:8]: Yeah it's pretty weird, potentially a bug that I could patch however the package doesn't look that active so for now worth just getting the types aligned to the current response. We saw a bunch of failures in prod due to this overnight 😅

**Comment by viralpickaxe** [comment]
Ready to merge

**Pull request merged**